Sources/SentrySwiftUI/SentryPerformanceView.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import Sentry
+import SwiftUI
+#if SWIFT_PACKAGE
+import SentryInternal
+#endif
+
+/// A control to measure performance of you views and send the result as transaction to Sentry.io.
+///
+/// You create a transaction by wrapping your views with this.
+///
+///     SentryPerformanceView {
+///         VStack {
+///             //The part of your content you want to measure
+///         }
+///     }
+///
+/// By default the transaction name will be the first root view, in the case above `VStack`,
+/// you can give your transaction a custom name by providing the name parameter.
+///
+///     SentryPerformanceView("My Awesome Screen") {
+///         VStack {
+///             //The part of your content you want to measure
+///         }
+///     }
+///
+/// Alternatively you can use the view extension:
+///
+///     VStack {
+///         //The part of your content you want to measure
+///     }.sentryTransaction("My Awesome Screen")
+///
+///
+@available(iOS 13, macOS 10.15, tvOS 13, watchOS 6.0, *)
+public struct SentryPerformanceView<Content: View>: View {
```

Comment:
`m`: We have some UITests for this class, but I think adding some unit tests would also be great. We could move all the logic to an internal class and write some unit tests for it.

Reply:
Most logics in here are related to SwiftUI events, I believe the best way to test this is in a UI test, unit tests would be just redundant, I could create a test for `extractName`, but it does not seems worth to have a unit test just for this, we are also testing its result in the UI test.